contents/blog/clickhouse-materialized-columns.md

```diff
@@ -0,0 +1,134 @@
+---
+date: 2021-09-09
+title: Speeding up ClickHouse with Materialized Columns
+rootPage: /blog
+sidebar: Blog
+showTitle: true
+hideAnchor: true
+categories: engineering
+author: karl-aksel-puulmann
+featuredImage: ../images/blog/automating-software-company-github-actions.png
+featuredImageType: full
+---
+
+Did you know ClickHouse supports speeding up queries up to an order of magnitude by using Materialized Columns? This rarely-used feature can be used to create new columns on the fly from existing data, speeding up queries.
```

Comment:
I think we can expand on this quite a bit with 1. what is PostHog, 2. What is Clickhouse, 3. why do we use clickhouse. (maybe one for @shehuphd )

It would also be great to tell the story of what issues we were running against, the stuff we tried that didn't work, how we ended up going for mat columns etc.

Reply:
I think all of these are deserving of a blog post/talk, but doesn't belong here. It's also partially covered by https://posthog.com/blog/clickhouse-announcement

The intent is to cover a niche area of ClickHouse with some amount of depth - it's the blog post I'd have wanted setting out on this.

Reply:
Depends on the target audience, but in case this seems to be for engineers that know ClickHouse who want to make the most of it, in which case this is a good premise.

Nit: "speeding up queries" is used twice in this paragraph, maybe the second occurrence could be worded differently?

Reply:
Tweaked the intro for greater conciseness.

Reply:
@shehuphd Still missing a tiny bit of intro on PostHog etc